### DISCUSSION TITLE
Fastify hot-reload during development

### DISCUSSION THREAD

**Discussion by hugoattal** [fastify/fastify#3795] [Q&A] [answered]
So, I know that Fastify kinda "compiles" route and plugins for fast access before listening to request.
And I absolutely love Fastify for that, as I want a fast backend in production.

However, it can be a bit frustrating during development. I don't really care that response are fast or not when I'm developing it in local.
For now, I use [nodemon](https://www.npmjs.com/package/nodemon) to restart my local Fastify server whenever I make a change. The thing is that's reaaaally slow. My server used to take more than a minute to get started, I optimized it a bit to come down to nearly 15s, but it's still a bit painful.

I wanted to know if you guys had an idea/solution/anything to make hot reload work during development? (like, in less than a second)
No worry if that's not possible, I'll 100% keep using Fastify, but just wanted to give it a shot.

(In any case, thanks so much to the whole team, you guys rock ❤️)

**Comment by zekth** [comment]
What kind of setup do you have? Js or Ts? I personnaly use nodemon with TS and it's like 10s-15s. Which is acceptable regarding the stack involved.

**Reply by hugoattal** [reply]
TS (and I use ts-node as CLI). Yes, 10-15s is acceptable, I'm just looking for a fast hot-reload. I'm using vite in the frontend, and I'm getting used to instant reload.
It's really just for comfort, it's not essential, but wanted to know if anyone had a solution for that.

**Reply by zekth** [reply]
`incremental` build also helps a lot and seems to be supported in `ts-node` now.

**Reply by mcollina** [reply]
I have been experimenting with 
https://github.com/fastify/restartable and https://github.com/mcollina/fastify-isolate for similar reasons. It would be awesome to combine these and typescript to automatically recompile and reload.

**Reply by meotimdihia** [reply]
@mcollina hopefully, we can release it soon. It is kind of suck when the application is becoming bigger.

**Reply by mcollina** [reply]
If you are looking for this kind of flow, you can try [platformatic service](https://oss.platformatic.dev/docs/reference/service/introduction). Automatic live reload with TS recompilation.

**Comment by hugoattal** [comment, marked as answer]
I just add this message in case someone is looking to reduce startup speed during development:

You can just disable type checking (which remains active on your IDE of course), and it will **drastically** reduce startup time.
For this, just use the `transpileOnly` flag (shortcut `T`) on ts-node or nodemon.

Example:
```bash
> ts-node --transpileOnly src/index.ts
> ts-node -T src/index.ts
```

cf https://typestrong.org/ts-node/docs/options/#transpileonly

**Reply by twibster** [reply]
Not all heroes wear capes
